src/cc/usdt/usdt.cc

```diff
@@ -193,40 +212,87 @@ bool Probe::usdt_getarg(std::ostream &stream) {
   return true;
 }
 
+uint64_t Probe::usdt_getargfmt() {
+  assert(!locations_.empty());
+  if (argfmt_ == (uint64_t)-1) {
+    argfmt_ = 0;
+    const size_t arg_count = locations_[0].arguments_.size();
+
+    if (arg_count > 0) {
+      assert(arg_count <= 16);
+      int t = largest_arg(0)->arg_size();
+      assert(t > -5 && t < 5);
+      if (t < 0) {
+        argfmt_ = 1 | (-t);
+      } else {
+        argfmt_ = t;
+      }
+      for (size_t arg_n = 1; arg_n < arg_count; ++arg_n) {
+        argfmt_ <<= 4;
+        int t = largest_arg(arg_n)->arg_size();
+        assert(t > -5 && t < 5);
+        if (t < 0) {
+          argfmt_ |= (1 | (-t));
+        } else {
+          argfmt_ |= t;
+        }
+      }
+    }
+  }
+  return argfmt_;
```

Comment:
Some comments are needed to understand the algorithm ...

Reply:
OK. Ya, it's basically an encoding of the argument types in a single 64 bit value. I'll clarify.